webapp/src/main/java/io/github/microcks/web/SoapController.java

```diff
@@ -115,6 +116,11 @@ public ResponseEntity<?> execute(
       log.info("Service name: " + serviceName);
       // Retrieve service and correct operation.
       Service service = serviceRepository.findByNameAndVersion(serviceName, version);
+      if (service == null) {
+         return new ResponseEntity<String>(
+            String.format("The service %s with version %s does not exist!", serviceName, version),
+            HttpStatus.NOT_FOUND);
```

Comment:
We've got a specific discussion on this topic here: https://github.com/microcks/microcks/issues/819
Can you check and join the party?

Reply:
Yes, I agree with this one but it's to get a global view on which status code to use depending on the failure case.